### PR TITLE
Add the dynamic shape unit test the CI.

### PR DESCRIPTION
**Pull request by vanbasten23**
Fix [#ISSUE_NUMBER](https://github.com/pytorch/xla/issues/4171).

### CODE REVIEW

File: torch_xla/csrc/tensor.h [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 namespace torch_xla {
 
-class TORCH_API XLASymNodeImpl : public c10::SymNodeImpl {
+class TORCH_API XLASymNodeImpl : public torch::impl::PythonSymNodeImpl {
```
Comment (ezyang) [inline torch_xla/csrc/tensor.h:30]: this should not be necessary

File: test/run_tests.sh [hunk 2]
```diff
@@ -108,6 +108,7 @@ function run_op_tests {
   run_dynamic python3 "$CDIR/../../test/nn/test_embedding.py" "$@" -v TestEmbeddingNNDeviceTypeXLA
   run_dynamic python3 "$CDIR/../../test/test_type_promotion.py" "$@" -v TestTypePromotionXLA
   run_dynamic python3 "$CDIR/test_operations.py" "$@" --verbosity=$VERBOSITY
+  run_dynamic python3 "$CDIR/test_dynamic_shapes.py" "$@"
```
Comment (ezyang) [inline test/run_tests.sh:111]: Don't you need to turn on experimental shapes

Comment (JackCaoG) [reply test/run_tests.sh:111]: `run_dynamic` will enable those flags. 
https://github.com/pytorch/xla/blob/515d53e9d78b5788e3f0b351b50b9b4423d0e80e/test/run_tests.sh#L64-L71

File: torch_xla/csrc/tensor.h (hunk 1 above)
Comment (vanbasten23) [reply torch_xla/csrc/tensor.h]: reverted.

**Comment by vanbasten23** [comment]
> 

Yes, the test start passing.

**Review by JackCaoG — APPROVED** [review]
Thanks! Feel free to merge once CI is green.

**Pull request merged**